README.md

```diff
@@ -16,6 +35,280 @@ ember install @bagaar/ember-permissions
 
 ## Usage
 
+### 1\. Setting and Checking Permissions for the Current Session
+
+First, we need to let the `permissions` service know which permissions are available for the current session. In the example below, we're using an additional service to request the permissions from a server API. Afterwards, we pass along the permissions to the `permissions` service via the `setPermissions` method.
+
+```javascript
+// app/routes/application.js
+
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default Route.extend({
+  apiService: service('api'),
+  permissionsService: service('permissions'),
+
+  async beforeModel() {
+    let permissions = await this.apiService.request('/permissions');
+
+    this.permissionsService.setPermissions(permissions);
+  }
+});
+```
+
+Once the permissions are set, we can start checking their presence. In the example below, we use the `has-permissions` helper to conditionally render a button based on the presence of a specific permission.
+
+```handlebars
+{{! app/templates/users/index.hbs }}
+
+{{#if (has-permissions "delete-users")}}
+  <button onclick={{action this.deleteUser userRecord}} type="button">
+    Delete User
+  </button>
+{{/if}}
+```
+
+> **NOTE:** If you need to check permissions inside a JavaScript file, you can use the `hasPermissions` method on the `permissions` service instead.
+
+### 2\. Setting the Required Permissions per Route, Watching Transitions and Checking Route Access
+
+Start of with defining the required permissions per route. You're free to define them where you want, as long as the format is the same as shown below.
+
+```javascript
+// app/route-permissions.js
+
+export default {
+  'users.index': ['view-users'],
```

Comment:
maybe change this to `'route.name': ['permission-name']` to make it explicit what the strings are (even though it's fairly obvious of course..)